src/tribler/gui/start_gui.py

```diff
@@ -38,8 +38,12 @@ def run_gui(api_port, api_key, root_state_dir, parsed_args):
         logger.info('Enabling a workaround for Ubuntu 21.04+ wayland environment')
         os.environ["GDK_BACKEND"] = "x11"
 
-    # Set up logging
-    load_logger_config('tribler-gui', root_state_dir)
```

Comment:
As it turns out, the previous code contains a bug. Tribler Core/GUI processes should not load logger config before determining whether the current process is the primary or not. Otherwise, primary and secondary processes start writing to log files, and the logging module does not support writing to the same log file from multiple processes. It leads to random errors during the log rotation, like, using an incorrect file descriptor.

In this PR, I fix the problem by first checking whether the current process is primary and then initializing file-based logging for a primary process only.